src/components/StageInfo.tsx

```diff
@@ -0,0 +1,43 @@
+// Cleanup CSS!
+import '../App.css';
+import { Component } from 'react';
+import * as React from 'react';
+
+export default class StageInfo extends Component<any, any> {
+  constructor(props: any, context: any) {
+    super(props, context);
+
+  }
+
+  render() {
+    const numSelected = this.props.cards.filter( (c: any) => c.mark == this.props.stage).length;
+    const needed = this.props.cardsNeeded[this.props.stage]
+
+    let button = null
+    if (numSelected == needed) {
+      button = <span
```

Comment:
General preferred pattern here is to define the rendering inline with the return if possible. That makes the below code:

```
{
    numselected === needed ? (
        <button stuff... />
    ) : null
}
```